ARM64:dts:aspeed Morocco DTS changes for A1

aasboddu opened the pull request:
Update DTS entries according to change in A1 Aspeed BMC chip. Following are the changes.
- Add LTPIG GPIO/UART mapping based on A1
- Update vBIOS/xDMA mapping for KVM 
- BIOS/FPGA flash change to SPI2 controller
- I2C Bus 10 change to I2C Bus 12
- USB VHUB change to XHCI
- eSPI GPIO change for POST Codes

tested: verified in Morocco platform

aasboddu committed:
ARM64:dts:aspeed:Morocco: Fix memory map

Fix node name in memory mapping for A1 Silicon

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: add vBIOS/xDMA for KVM

Add vBIOS and xDMA memory to enable KVM for A1
silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add BIOS/FPGA flash memory

Add SPI2 controller for BIOS/FPGA flash memory
access. Updated clock speed for SPI devices to 33MHz.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Modify I2C Bus for Morocco A1

Modify I2C Bus number for Morocco based on
A1 silicon specification.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add A1 LTPI GPIO mapping

Add LTPI GPIO mapping based A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add A1 LTPI UART mapping

Add LTPI UART mapping based A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add USB vHUB for KVM

Add USB vHUB to enable keyboard/mouse for KVM

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add SGPIO for A1 silicon

Add SGPIO support for A1 silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

ojayanth commented:
Please provide summary in PR about the changes.

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -65,16 +65,33 @@
-		pcc_memory: pccbuffer {
-                        reg = <0xE0000000 0x00001000>; /* 4K */
-			no-map;
+                pcc_memory: pccbuffer {
```
No changes here , revert the formating chnages.

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -40,6 +40,21 @@
 			no-map;
 		};
 
+                espi_mmbi_memory: espi-mmbi-memory@428000000 {
```
fix  formating issues

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -550,7 +550,7 @@
 	};
 };
 
-&i2c10 {
```
Replace Congo name to morocco in commit message.

aasboddu commented:
> Please provide summary in PR about the changes.

Provided summary for list of changes in the PR.

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add GPIO HOG for eSPI

Add GPIO HOG to set eSPI GPIO line to low and
direction to output enabling BMC to receive BIOS
POST codes for A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -229,12 +229,41 @@
+                reg = <1>;
+                status = "okay";
+                m25p,fast-read;
+                label = "pnor";
```
Label should be diffrent?

aasboddu replied:
CS1 removed as it is not used.

mahkurap commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -972,6 +972,12 @@ spd_ ## bus ## _ ## index: spd@addr,4cc5118 ## index ## 000 { \
 
 &gpio0 {
         status = "okay";
+        espi_default {
```
Please add tabs instead of spaces

srgovard commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -40,6 +40,21 @@
 			no-map;
 		};
 
+		espi_mmbi_memory: espi-mmbi-memory@428000000 {
```
As per ast2700 dts (evn, reservedmemory) files this has defined espi0 and espi1. Above address belongs to espi1. Please check with Raja/Ashok on this.

aasboddu replied:
Checked with ashok/raja no change here.

srgovard commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -65,17 +65,34 @@
-			no-map;
+                pcc_memory: pccbuffer {
+                        reg = <0x4 0xE0000000 0x00001000>; /* 4K */
+                        no-map;
```
Address the space issue.

aasboddu replied:
ACK

srgovard commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -1045,14 +1045,21 @@ spd_ ## bus ## _ ## index: spd@addr,4cc5118 ## index ## 000 { \
-       /delete-property/ pinctrl-0;
-       status = "okay";
+&uart13 {
+	/delete-property/ pinctrl-names;
```
Address space issue here.

aasboddu replied:
ACK

srgovard commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -1020,6 +1020,10 @@ spd_ ## bus ## _ ## index: spd@addr,4cc5118 ## index ## 000 { \
 		/*216-223*/ "","","","","","","","";
 };
 
+&sgpios {
```
Are we using sgpios?

aasboddu replied:
Added from working congo dts. Will clarify with ashok

srgovard commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-morocco.dts:
```diff
@@ -1037,19 +1043,15 @@ spd_ ## bus ## _ ## index: spd@addr,4cc5118 ## index ## 000 { \
+	memory-region = <&pcc_memory>;
+	rec-mode = <0x1>;
+	port-addr-hbits-select = <0x1>;
+	port-addr-xbits = <0x3>;
```
Fix these.. no changes needed here.

aasboddu replied:
added tab space for consistency
